src/runtime/particle-execution-context.ts

```diff
@@ -57,17 +59,22 @@ export class ParticleExecutionContext implements StorageCommunicationEndpointPro
 
     this.apiPort = new class extends PECInnerPort {
 
-      onDefineHandle(identifier: string, type: Type, name: string) {
+      onDefineHandle(identifier: string, type: Type, name: string, isNG: boolean) {
+        if (isNG) {
```

Comment:
I'd suggest just checking Flags instead of this. Otherwise we have to add a whole lot of plumbing that we'll later need to remove again.

Reply:
Done